cli/dcoscli/tables.py

```diff
@@ -849,10 +850,31 @@ def slave_table(slaves):
         ('ID', lambda s: s['id'])
     ])
 
-    tb = table(fields, slaves, sortby="HOSTNAME")
+    for field_name in field_names:
+        if field_name.lower() == 'ip':
+            fields[field_name.upper()] = lambda s: mesos.parse_pid(s['pid'])[1]
+        else:
+            if ':' in field_name:
+                heading, field_name = field_name.split(':', 1)
+            else:
+                heading = field_name
+            fields[heading.upper()] = _dotted_itemgetter(field_name.lower())
+
+    sortby = list(fields.keys())[0]
+    kwargs = {}
+    if sortby.lower() == 'ip':
+        kwargs['sort_key'] = lambda d: tuple(int(x) for x in d[0].split('.'))
+    tb = table(fields, slaves, sortby=sortby, **kwargs)
```

Comment:
Lets remove it. Not a fan of having unnecessary code, tends to just cause confusion down the road.